In [4]:
import re

pattern = r"(FROM:(?P<FromCity>[\w\s\/]*)\s*,\s*(?P<FromCountry>[\w\s\/]*)TO:(?P<ToCity>[\w\s\/]*)\s*,\s*(?P<ToCountry>[\w\s\/]*)(?P<ToAirport>[A-Z]{3})ValidityDaysDepTimeArrTimeFlightAircraftTravelTime)*(?P<Validity>\d\d\s\w{3}\s*-\s*\d\d\s\w{3})(?P<Days>[\d\s]*)(?P<DepTime>\d\d:\d\d)(?P<ArrTime>\d\d:\d\d[+1]*)(?P<Flight>[A-Z]{2}\d{3,4})\**(?P<Aircraft>[A-Z\d]{3})(?P<TravelTime>\d{1,2}H\d*M)"

with open('text_pdf.txt', 'r') as pdf:
    text = pdf.readlines()
    



In [40]:
with open('pdf_all.csv', 'w') as file:
    count = 1
    for line in text:
        match = re.match(pattern, line)
        if match != None:
            print(match.group('FromCity'))
            print(match.group('FromCountry'))
            print(match.group('ToCity'))
            print(match.group('ToCountry'))
            print(match.group('ToAirport'))
            print(match.group('Validity'))
            print(match.group('Days'))
            print(match.group('DepTime'))
            print(match.group('ArrTime'))
            print(match.group('Flight'))
            print(match.group('Aircraft'))
            print(match.group('TravelTime'))

            
            splt = re.split(pattern, line)
            print('---')
            for i in range(2, 7): #City - similar for page
                print(splt[i])
            print('---')
            for i in range(7, 14):
                print(splt[i])
            print('---')
            for i in range(21, 28):
                print(splt[i])
            print('---')    
            for i in range(35, 42):
                print(splt[i])
            print('---')
            print(len(splt))
            count += 1
            if(count == 10):
                break

Aalborg
DenmarkAAL
Amsterdam 
Netherlands
AMS
01 Nov  -  31 Jan
1234567
06:00
07:25
KL1328
73W
1H25M
---
Aalborg
DenmarkAAL
Amsterdam 
Netherlands
AMS
---
01 Nov  -  31 Jan
1234567
06:00
07:25
KL1328
73W
1H25M
---
01 Nov  -  31 Jan
1234567
12:10
13:35
KL1334
73W
1H25M
---
01 Nov  -  23 Dec
1234567
18:15
19:35
KL1336
EQV
1H20M
---
1289
None
None
None
None
None
01 Nov  -  31 Jan
1234567
12:10
13:35
KL1334
73W
1H25M
---
None
None
None
None
None
---
01 Nov  -  31 Jan
1234567
12:10
13:35
KL1334
73W
1H25M
---
01 Nov  -  23 Dec
1234567
18:15
19:35
KL1336
EQV
1H20M
---
26 Dec  -  06 Jan
1  34567
18:15
19:35
KL1336
EQV
1H20M
---
113
Aalesund 
NorwayAES
Amsterdam 
Netherlands
AMS
01 Nov  -  31 Jan
1  345  7
16:50
18:55
KL1326
E75
2H5M
---
Aalesund 
NorwayAES
Amsterdam 
Netherlands
AMS
---
01 Nov  -  31 Jan
1  345  7
16:50
18:55
KL1326
E75
2H5M
---
02 Nov  -  28 Jan
  2      6
17:10
19:15
KL1326
E75
2H5M
---
01 Nov  -  31 Jan
1  345  7
16:50
18:55
KL1326
E75
2H5M
---
561
Abakan
Russian Federation

In [34]:
steps = int((len(splt) - 1) / 14)
for step in range(0, steps):
    for i in range(7 + 14 * step, 14 + 14 * step):
        print(splt[i])
    print('\n')

01 Nov  -  31 Jan
1234567
06:00
07:25
KL1328
73W
1H25M


01 Nov  -  31 Jan
1234567
12:10
13:35
KL1334
73W
1H25M


01 Nov  -  23 Dec
1234567
18:15
19:35
KL1336
EQV
1H20M


26 Dec  -  06 Jan
1  34567
18:15
19:35
KL1336
EQV
1H20M


07 Jan  -  31 Jan
1234567
18:15
19:35
KL1336
EQV
1H20M


01 Nov  -  31 Jan
1234567
06:00
07:25
KL1328
73W
1H25M


01 Nov  -  31 Jan
1234567
12:10
13:35
KL1334
73W
1H25M


01 Nov  -  23 Dec
1234567
18:15
19:35
KL1336
EQV
1H20M


26 Dec  -  06 Jan
1  34567
18:15
19:35
KL1336
EQV
1H20M


07 Jan  -  31 Jan
1234567
18:15
19:35
KL1336
EQV
1H20M


01 Nov  -  31 Jan
1234567
06:00
07:25
KL1328
73W
1H25M


01 Nov  -  31 Jan
1234567
12:10
13:35
KL1334
73W
1H25M


01 Nov  -  23 Dec
1234567
18:15
19:35
KL1336
EQV
1H20M


26 Dec  -  06 Jan
1  34567
18:15
19:35
KL1336
EQV
1H20M


07 Jan  -  31 Jan
1234567
18:15
19:35
KL1336
EQV
1H20M


01 Nov  -  31 Jan
1234567
06:00
07:25
KL1328
73W
1H25M


01 Nov  -  31 Jan
1234567
12:10
13:35
KL1334
73W
1H25M


01 Nov  -  23 Dec
1234567
18

In [41]:
for i in range(2, 7): #City - similar for page
    print(splt[i])

Aberdeen 
United KingdomABZ
Paris 
France
CDG


In [36]:
header_string = 'FromCity,FromCountry,ToCity,ToCountry,ToAirport,Validity,Days,DepTime,ArrTime,Flight,Aircraft,TravelTime\n'

##Match 1
##Group `FromCity`	5-12	`Aalborg`
##Group `FromCountry`	14-24	`DenmarkAAL`
##Group `ToCity`	27-37	`Amsterdam `
##Group `ToCountry`	39-50	`Netherlands`
##Group `ToAirport`	50-53	`AMS`
##Group `Validity`	103-120	`01 Nov  -  31 Jan`
##Group `Days`	120-127	`1234567`
##Group `DepTime`	127-132	`06:00`
##Group `ArrTime`	132-137	`07:25`
##Group `Flight`	137-143	`KL1328`
##Group `Aircraft`	143-146	`73W`
##Group `TravelTime`	146-151	`1H25M`

##Match 2
##Full match	151-199	`01 Nov  -  31 Jan123456712:1013:35KL133473W1H25M`
##Group `Validity`	151-168	`01 Nov  -  31 Jan`
##Group `Days`	168-175	`1234567`
##Group `DepTime`	175-180	`12:10`
##Group `ArrTime`	180-185	`13:35`
##Group `Flight`	185-191	`KL1334`
##Group `Aircraft`	191-194	`73W`
##Group `TravelTime`	194-199	`1H25M`